File: node/JSON2Excel.js
const fs = require('fs');
const ExcelJS = require('exceljs');

// Configuration parameters
const inputFileName = 'input.json'; // Name of the input JSON file
const date = new Date().toDateString(); // Timestamp of output Excel file
const outputFilePath = './node/files/'; // File path where output Excel file will be written to
const outputFileName = outputFilePath + 'output ' + date + '.xlsx'; // Name of the output Excel file
const sheetName = 'News For ' + date; // Name of the sheet in the Excel file
const data = require(`../${inputFileName}`); // JSON data structure retrieved from input JSON file
const dataObject = data['articles']; // JSON data structure to be parsed and written to the Excel file
const headersObject = data['articles'][0]; // JSON data structure of the headers to be written to the Excel file

// Function to convert JSON data to Excel
async function convertJsonToExcel() {
    // Create a new Excel workbook
    const workbook = new ExcelJS.Workbook();
    const worksheet = workbook.addWorksheet(sheetName);

    // Write headers
    const headers = Object.keys(headersObject);
    worksheet.addRow(headers);

    // Write data
    dataObject.forEach(obj => {
        const row = [];
        headers.forEach(header => {
            row.push(obj[header]);
        });
        worksheet.addRow(row);
    });

    // Save the workbook
    await workbook.xlsx.writeFile(outputFileName);
    console.log(`Excel file "${outputFileName}" generated successfully.`);
}

// Main function
async function main() {
    try {
        await convertJsonToExcel();
    } catch (error) {
        console.error('[JSON2Excel] Error:', error.message);
    }
}

// Execute the main function
main();

File: python/JSON2Excel.py
import json
import datetime
from openpyxl import Workbook

# Configuration parameters
input_file_name = 'input.json'  # Name of the input JSON file
date = datetime.datetime.now() # Timestamp of output Excel file
output_file_path = './python/files/' # File path where output Excel file will be written to
output_file_name = output_file_path + 'output ' + date.strftime("%Y-%m-%d") + '.xlsx'  # Name of the output Excel file
sheet_name = 'News For ' +  date.strftime("%Y-%m-%d");  # Name of the sheet in the Excel file

# Function to read JSON data from file
def read_json(file_name):
    with open(file_name, 'r') as f:
        return json.load(f)

# Function to write JSON data to Excel
def write_excel(data):
    workbook = Workbook()
    worksheet = workbook.active
    worksheet.title = sheet_name
    data = data['articles']

    # Write headers
    headers = list(data[0].keys())
    worksheet.append(headers)

    # Write data
    for item in data:
        row = [str(item[header]) for header in headers]
        worksheet.append(row)

    # Save the workbook
    workbook.save(output_file_name)
    print(f'Excel file "{output_file_name}" generated successfully.')

# Main function
def main():
    try:
        # Read JSON data from file
        json_data = read_json(input_file_name)
        # Write JSON data to Excel
        write_excel(json_data)
    except Exception as e:
        print('Error:', e)

# Execute the main function
# if __name__ == "__main__":
main()
